lib/rexml/parsers/baseparser.rb

```diff
@@ -460,15 +461,24 @@ def pull_event
                 @closed = tag
                 @nsstack.shift
               else
+                if @tags.size.zero? && !@root_tag.nil?
+                  raise ParseException.new("Malformed XML: Extra tag at the end of the document (got '<#{tag}')", @source)
+                end
                 @tags.push( tag )
               end
+              @root_tag ||= tag
```

Comment:
How about using boolean for root element check because we don't need root element name?

```suggestion
                if @tags.empty?
                  if @have_root
                    raise ParseException.new("Malformed XML: Extra tag at the end of the document (got '<#{tag}')", @source)
                  end
                  @have_root = true
                end
                @tags.push( tag )
              end
```

Reply:
OK.
I see.
Thanks!